### ISSUE TITLE
Import into react-native is broken

### ISSUE THREAD

**Issue by vanjaoljaca** [adzejs/adze#180]
I'm having trouble getting this to load into a mobile app, I think it is incorrectly detecting the environment and using some incompatible libraries. 

```
npx create-expo-app MyAwesomeApp
cd MyAwesomeApp
npx expo install adze
```

I tried to workaround with polyfills but couldn't get it working

**Comment by AJStacy** [comment]
So, after investigation, the issue is related to commonjs vs esm modules. Chalk 5, which is what Adze uses by default is ESM only. Expo tries to bundle Adze using commonjs, and then explodes when it tries to bundle chalk. There is no solution to resolving this issue directly in Adze. The solution is to add a small overrides declaration in package.json to use Chalk v4 vs the default v5 that comes with Adze.

```json
{
  "overrides": {
     "adze": {
       "chalk": "^4.1.2"
     }
  }
}
```

I will be putting these directions in the documentation under the FAQ's section which should solve this issue for expo users.

**Comment by AJStacy** [comment]
@vanjaoljaca , why did you need to comment this code block out?

```
/**
 * Validates that the current environment is `Window`.
 */
export function isBrowser() {
    return false;
    // return typeof window !== 'undefined' && !isDeno();
}
```

What issue did that resolve for you?

**Comment by vanjaoljaca** [comment]
without that change I get errors because _glbl.navigator.userAgent is undefined

![image](https://github.com/user-attachments/assets/f19b8361-ae7b-4b4e-b7a5-c755138f2193)

global.js:75
```
/**
 * Validates the current environment is Firefox.
 */
export function isFirefox() {
    const _glbl = globalThis;
    if (envIsWindow(_glbl)) {
        return _glbl.navigator.userAgent.includes('Firefox');
    }
    return false;
}
```
constants.js:173
```
/**
 * Default log configuration for warn logs.
 */
export function getWarnConfig(overrides = {}) {
    return {
        levelName: 'warn',
        level: 2,
        style: `font-size: 12px; border-radius: 4px;  background: linear-gradient(to right, #fff, #fff0a8); color: #715100; border-color: #e3d696; padding-right: ${isFirefox() ? '44px' : '30px'};`,
        terminalStyle: ['white', 'bgYellow'],
        method: 'warn',
        emoji: '🔔',
        ...overrides,
    };
}
```

**Comment by AJStacy** [comment]
Thank you. That's helpful. Expo is somehow thinking `window` exists when it doesn't. I'll see if I can replicate that issue and resolve it.

**Comment by vanjaoljaca** [comment]
ah yes I see envIsWindow maps to isBrowser, which is how I ended up making isBrowser return false

```
export function envIsWindow(_) {
    return isBrowser();
}
```

**Comment by tiagobnobrega** [comment]
Hi... I was experimenting with picocolors instead of Chalk to address some of the issues I found. I also spotted the `isBrowser` check issue. I created this PR to replace Chalk and fix some of these issues. Not sure that using piccocolors is the direction you want to take, but The PR should at least be insightful to some of the issues when running on Hermes/Expo

Here's the link to the PR: https://github.com/adzejs/adze/pull/181
